src/service/resources-impl.js

```diff
@@ -1741,28 +1816,57 @@ export class Resource {
   }
 
   /**
-   * Calls element's documentInactiveCallback callback and resets state for
+   * Calls element's unlayoutCallback callback and resets state for
    * relayout in case document becomes active again.
    */
-  documentBecameInactive() {
-    if (this.state_ == ResourceState_.NOT_BUILT) {
+  unlayout() {
+    if (this.state_ != ResourceState_.LAYOUT_COMPLETE) {
       return;
     }
     if (this.isInViewport()) {
       this.setInViewport(false);
     }
-    if (this.element.documentInactiveCallback()) {
+    if (this.element.unlayoutCallback()) {
       this.state_ = ResourceState_.NOT_LAID_OUT;
       this.layoutCount_ = 0;
     }
   }
 
   /**
-   * Called when a previously visible element has become invisible.
+   * Calls element's pauseCallback callback.
+   */
+  pause() {
+    if (this.state_ == ResourceState_.NOT_BUILT) {
+      return;
+    }
+    if (this.isInViewport()) {
+      this.setInViewport(false);
+    }
+    this.element.pauseCallback();
+  }
+
+  /**
+   * TODO(jridgewell): This should eventually call layoutCallback
+   */
+  layout() {
+  }
+
+  /**
+   * Calls element's resumeCallback callback.
+   */
+  resume() {
+    if (this.state_ == ResourceState_.NOT_BUILT) {
+      return;
+    }
+    this.element.resumeCallback();
+  }
+
+  /**
+   * Called when a previously visible element is no longer displayed.
    */
   unload() {
-    // TODO(dvoytenko): Likely warrants its own callback and re-layout rules.
-    this.documentBecameInactive();
+    this.pause();
```

Comment:
What I mean is that the `unload` method here does both: `pause` AND `layout`. However, above in the transitions, you have separately:
```
const pause = () => {
  this.resources_.forEach(r => r.pause());
};
const unlayout = () => {
  this.resources_.forEach(r => r.unlayout());
};
```
and also
```
const pauseAndUnlayout = () => {
  this.resources_.forEach(r => {
    r.pause();
    r.unlayout();
  });
};
```

So, my point was that this is either: (a) a duplicative code; (b) a bug; (c) me not understanding it correctly.